core/src/event_id.rs

```diff
@@ -141,6 +141,24 @@ impl EventId {
             cid,
         })
     }
+
+    /// swap_cid replaces the event CID in the EventId with a new CID.
+    pub fn swap_cid(&self, cid: &Cid) -> Option<Self> {
```

Comment:
This seems a bit odd.. do we actually need it (e.g. can we just call `EventId::new(..old values.., new_cid)`) or is there a way to use the builder to make this more natural?

If so, should we consume the EventId (self) or does it need to stay around? It seems like this should be taking a previously valid EventId and returning a transformed one rather than keeping both around.

Reply:
So we need to have the new time event know the recon EventID so we can construct an event insertable. What we have is the recon EventID of the data event being anchored.  To use the builder, we would need information that is not in the Events table. e.g. network ID, separator key, separator value, controller. 
We have two options. Crate a builder that circumvents the protections and lets you just construct with raw values, deconstruct a event id to parts, send the parts around and recombine the parts or jsut send the eventID and have an update CID.

I picked this method as it keeps the structure of the eventID opaque to the anchor service. All it knows it 
```rs
f(data event eventID, time event CID) -> time event EventID
```